### PR TITLE
feat(mesh): Add mesh config and others

### PR DESCRIPTION
**Pull request by angelsvirkov**
- Add `graphql-tag` package
- Add mesh config
- Add additional type defs / resolvers
- Add user rights
- Add custom fetch
- Add `getUserRights` middleware

### CODE REVIEW

File: packages/backend/src/config/Endpoints.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const Endpoints = {
+  // TODO: Find a Public GQL Service which can be used in two environments way
```
Comment (Kiesen) [inline packages/backend/src/config/Endpoints.ts:2]: I think finding a public GQL service with two environments is hard to find. Maybe we can just use one service but within two input handlers? 

This one: https://graphqlzero.almansi.me/ looks promising to me.

File: packages/backend/src/mesh/mesh.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import { GraphQLSchema } from 'graphql';
+import GraphQLHandler from '@graphql-mesh/graphql';
+import MySQLHandler from '@graphql-mesh/mysql';
+import { MeshPubSub } from '@graphql-mesh/types';
+import { PubSub } from 'graphql-subscriptions';
+import LRUCache from '@graphql-mesh/cache-inmemory-lru';
+import StitchingMerger from '@graphql-mesh/merger-stitching';
+import FilterTransform from '@graphql-mesh/transform-filter-schema';
+import CacheTransform from '@graphql-mesh/transform-cache';
+import { additionalResolvers } from './additionalResolvers/additionalResolvers';
```
Comment (Kiesen) [inline packages/backend/src/mesh/mesh.ts:11]:
```suggestion
import { additionalResolvers } from '@src/mesh/additionalResolvers/additionalResolvers';
```

File: packages/backend/src/mesh/mesh.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import { additionalTypeDefs } from './additionalTypeDefs';
```
Comment (Kiesen) [inline packages/backend/src/mesh/mesh.ts:12]:
```suggestion
import { additionalTypeDefs } from '@src/mesh/additionalTypeDefs';
```

File: packages/backend/tsconfig.json [hunk 4]
```diff
@@ -44,7 +44,13 @@
 
     /* Module Resolution Options */
     "moduleResolution": "node" /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */,
-    // "baseUrl": "./" /* Base directory to resolve non-absolute module names. */,
+    "baseUrl": "./" /* Base directory to resolve non-absolute module names. */,
+    "paths": {
+      "@util/*": ["src/util/*"],
+      "@_types/*": ["src/_types/*"],
```
Comment (Kiesen) [inline packages/backend/tsconfig.json:50]: Just a minor but I would like to put the types under `types`

```suggestion
      "@types/*": ["src/types/*"],
```

**Review by Kiesen — CHANGES_REQUESTED** [review]
Looks good to me, I just found some minor things and recommendations.

File: packages/backend/tsconfig.json (hunk 4 above)
Comment (angelsvirkov) [reply packages/backend/tsconfig.json:50]: As agreed , we change it to
`@internalTypes/*: ["src/types/*]`

Comment (angelsvirkov) [reply packages/backend/tsconfig.json:50]: Done

File: packages/backend/src/app.ts [hunk 5]
```diff
@@ -1,6 +1,8 @@
 import express from 'express';
-// import graphqlHTTP from 'express-graphql';
-// import { getMeshConfig } from '@src/graphql/mesh/mesh';
+import graphqlHTTP from 'express-graphql';
+import { getMeshConfig } from '@src/mesh/mesh';
+import { getUserRights } from './middleware/getUserRights';
```
Comment (Kiesen) [inline packages/backend/src/app.ts:4]: Found a minor:

```suggestion
import { getUserRights } from '@src/middleware/getUserRights';
```

File: packages/backend/src/rights/rights.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
+import { ResolversComposition } from '@graphql-tools/resolvers-composition';
+import { MeshContext } from '@internalTypes/context';
+import { GraphQLFieldResolver } from 'graphql';
+import { MutationConfig } from './config';
```
Comment (Kiesen) [inline packages/backend/src/rights/rights.ts:4]: Minor:

```suggestion
import { MutationConfig } from '@src/rights/config';
```

File: packages/backend/src/rights/rights.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+type ResolverComposer = ResolversComposition<
+  GraphQLFieldResolver<any, MeshContext>
+>;
+
+/**
+ * When you look into mesh.ts you see that we're using a ResolversCompositionTransform to
+ * allow/deny mutations based on a user rights. This is configured via resolvers specifying
```
Comment (Kiesen) [inline packages/backend/src/rights/rights.ts:12]: Another minor:

```suggestion
 * allow/deny mutations based on user rights. This is configured via resolvers specifying
```

**Review by Kiesen — CHANGES_REQUESTED** [review]
Found three minors - after committing them the PR can get merged.

**Review by Kiesen — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**